packages/editor/src/components/provider/use-post-editor-layout.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+import { useSettings } from '@wordpress/block-editor';
+
+export function usePostEditorLayout( templatePostContentBlock ) {
```

Comment:
@tellthemachines I'm not entirely sure about the logic to compute the right layout in this hook. Maybe you can confirm, I just tried to copy the logic from the `VisualEditor` component of the edit-post package which I think you implemented (at least most of it)

Reply:
@tellthemachines might be able to confirm, but the logic looks good to me! (Attempts to use constrained if available, while accounting for legacy `inherit` property, and falls back to `default`).

Reply:
The logic is correct, but I don't think we need it because, unless I'm quite mistaken 😅, what we're doing here is just grabbing the Post Content layout from the template and applying it to a newly-created Post Content or Group block. 

The convoluted layering of fallbacks is necessary in the post editor because we have to manually determine the correct layout classnames to pass to BlockList so they get added at root level, and also add the correct layout styles for them to the page. Plus we're dealing with support for classic themes and all sorts of intermediate scenarios.

But if here we're just using blocks as containers for the post, the block editor already knows how to interpret their attributes, including the legacy ones 🙂 so we could just grab all the Post Content attributes from the template and add them directly instead, which would incidentally preserve any color or typography changes as @andrewserong commented above.